runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/EngineManager.java

```diff
@@ -430,7 +430,7 @@ private NameResolver(
             this.matchName = NamespaceAdapter.PATTERN_NAME.matcher("");
         }
 
-        private long resolve(
+        private synchronized long resolve(
```

Comment:
As a general strategy we tend to avoid locks, synchronized keyword, etc in the zilla codebase.

What is the race condition being solved by adding synchronized here? Is it reuse of `matchName` across multiple concurrent threads?

If so, then let's change `matchName` to be a `ThreadLocal`, and use the local matcher without introducing `syncrhonized`.

Reply:
> reuse of matchName across multiple concurrent threads
yep, exactly

>  let's change matchName to be a ThreadLocal
done, pls chk